File: Lecture20: ArrayQuestions/reverseArray.cpp
#include<iostream>
#include<vector>
using namespace std;

vector<int> reverse(vector<int> v) {

    int s = 0, e =v.size()-1;

    while(s<=e) {
        swap(v[s], v[e]);
        s++;
        e--;
    }

    return v;

}

void print(vector<int> v) {

    for(int i=0; i<v.size(); i++) {
        cout << v[i] << " ";
    }
    cout << endl;
}

int main() {

    vector<int> v ;

    v.push_back(11);
    v.push_back(7);
    v.push_back(3);
    v.push_back(12);
    v.push_back(4);

    vector<int> ans = reverse(v);

    cout << "Printing reverse Array" <<endl;
    print(ans);


    return 0;
}

File: Lecture020 ArrayQuestions/reverseArray.py

def reverse(v):
	s = 0
	e = len(v)-1

	while s <= e:
		v[s], v[e] = v[e], v[s]
		s += 1
		e -= 1

	return v


if __name__ == '__main__':
	v = []

	v.append(11)
	v.append(7)
	v.append(3)
	v.append(12)
	v.append(4)

	print(v)

	ans = reverse(v)
	print("Reverse Array")
	print(ans)
